hydrosdk/modelversion.py

```diff
@@ -307,16 +289,16 @@ def upload(self, cluster) -> dict:
             for metric in self.metrics:
                 upload_response = metric.model.__upload(cluster)
 
-                msc = MetricSpecConfig(model_version_id=upload_response.model.id,
+                msc = MetricSpecConfig(model_version_id=upload_response.modelversion.id,
                                        threshold=metric.threshold,
                                        threshold_op=metric.comparator)
                 ms_created = False
 
                 while not ms_created:
                     try:
-                        ms = MetricSpec.create(cluster=upload_response.model.cluster,
-                                               name=upload_response.model.name,
-                                               model_version_id=root_model_upload_response.model.id,
+                        ms = MetricSpec.create(cluster=upload_response.modelversion.cluster,
+                                               name=upload_response.modelversion.name,
+                                               model_version_id=root_model_upload_response.modelversion.id,
                                                config=msc)
                         ms_created = True
                     except MetricSpecException:
```

Comment:
Can we improve this code in this PR?
It's not clear why would we iterate regardless of the exception. What if we do not catch the exception and let it propagate?

Reply:
@KineticCookie resolved